packages/not-next/src/components/cart/Pay.tsx

```diff
@@ -62,10 +66,28 @@ export default function Pay({
         paymentArgs[0].currency !== abi.addresses.zeroAddress
       ) {
         debug("Pending ERC20 contract call approval");
-
+        const allowance = await getAllowance(shopPublicClient!, [
+          paymentArgs[0].payeeAddress,
+          paymentArgs[0].currency,
+        ]);
+        let amount = paymentArgs[0].amount;
+        if (allowance < paymentArgs[0].amount) {
+          amount = paymentArgs[0].amount - allowance;
+        }
```

Comment:
we should make at least one new variation of our tests where we trigger this code by manually setting a smaller allowance before.